src/ng/directive/ngAs.js

```diff
@@ -0,0 +1,148 @@
+'use strict';
+
+/**
+ * @ngdoc directive
+ * @name ngAs
+ * @restrict A
+ *
+ * @description
+ * The `ngAs` attribute tells Angular to assign element component controller
+ * to a given property.
+ *
+ * Using this directive you can use the controller of existing components
+ * in your template (children components).
+ *
+ * If the children component is destroyed
+ * a `null` is assigned to the property.
+ *
+ * Note that this is the reverse of `require:`:
+ * with `require:` is the children who references the parent
+ * but with `ngAs`is the parent who references the children.
+ * It is very useful when you want to reuse the same component
+ * in different situations,
+ * and they do not need to know which exact parent they have.
+ *
+ *
+ * @element ANY
+ * @param {expression} ngAs {@link guide/expression Expression} to assign the controller.
+ *
+ *
+ * @example
+ * ### Use inside the scope
+ * This example shows how the controller of the component toggle
+ * is reused in the template through the scope to use its logic.
+ * <example name="ngAsDirectiveComponentExample" module="ngAsExample">
+ *   <file name="index.html">
+ *     <toggle ng-as="myToggle"></toggle>
+ *     <button ng-click="myToggle.toggle()">Toggle</button>
+ *     <div ng-show="myToggle.isOpen()">You are using a children component to show it.</div>
```

Comment:
From this point of view yes.
Probably I should add a parent component to make it more explicit that it is the parent template who creates the relationships.